ui/src/views/Clients/ViewClient/index.jsx

```diff
@@ -84,20 +86,22 @@ export default class ViewClient extends Component {
   };
 
   handleDeleteClient = async clientId => {
-    this.setState({ error: null, loading: true });
+    this.setState({ dialogError: null, loading: true });
 
-    try {
-      await this.props.client.mutate({
-        mutation: deleteClientQuery,
-        variables: { clientId },
-      });
+    await this.props.client.mutate({
+      mutation: deleteClientQuery,
+      variables: { clientId },
+    });
```

Comment:
Remove the `async-await` stuff in this function and just use return. The `DialogAction` component will take care of running this function asynchronously. I think if you put `return this.props.client.mutate(...);` then the dialog will start showing a spinner above the "Delete Client" button.